src/nameutil.rs

```diff
@@ -125,6 +125,35 @@ pub fn lib_name_to_toml(name: &str) -> String {
     name.to_string().replace("-", "_").replace(".", "_")
 }
 
+pub fn shared_lib_name_to_link_name(name: &str) -> &str {
+    let mut s = name;
+
+    if s.starts_with("lib") {
+        s = &s[3..];
+    }
+
+    let offset = s.find(".so").unwrap_or_else(|| s.len());
+    s = &s[..offset];
+
+    s
+}
+
+pub fn shared_libs_to_links(shared_libs: &[String]) -> String {
+    let mut links = vec![];
+    for it in shared_libs {
+        links.push(format!("\"{}\"", shared_lib_name_to_link_name(it)));
+    }
+
+    if links.is_empty() {
+        panic!("empty list of shared library");
+    }
+
+    // https://github.com/rust-lang/cargo/issues/4533
+    // links = ["foo", "bar"] or ['foo', 'bar'] is not supported.
+    // Pick the first element
+    links[0].clone()
+}
```

Comment:
Considering that `shared_lib_name_to_link_name` always return something, why not replacing this function with:

```suggestion
pub fn shared_libs_to_links(shared_libs: &[String]) -> String {
    // https://github.com/rust-lang/cargo/issues/4533
    // links = ["foo", "bar"] or ['foo', 'bar'] is not supported.
    // Pick the first element
    for it in shared_libs {
        return format!("\"{}\"", shared_lib_name_to_link_name(it));
    }
    panic!("empty list of shared library");
}
```

?

Reply:
that looks much better :) thanks for the suggestion.

Reply:
Causes a clippy warning though, rightfully:

```
 src/nameutil.rs#L145

this loop never actually loops
```

Reply:
So instead of the loop, just do `if !shared_libs.is_empty() { ... }`

Reply:
hmm why my cargo build/test build doesn't tell warning? :) btw updated

Reply:
Run `cargo clippy` to get extra warnings/suggestions :)